### PR TITLE
[processor/k8sattributesprocessor] handle new pods

### PR DESCRIPTION
**Pull request by willthames**
**Description:** <Describe what has changed.>
Update start time check to ensure that new pods are included.

Without this fix, pods created after the opentelemetry collector
starts do not have their attributes set correctly

**Link to tracking Issue:** 

#13619 

**Testing:**

Ran this in kind before the fix and after the fix and validated that the pod info wasn't present following a restart before the fix but was present after the fix following a restart

**Documentation:**

No new documentation

### CODE REVIEW

File: processor/k8sattributesprocessor/internal/kube/client.go [hunk 1]
```diff
@@ -519,11 +519,11 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 
 	for _, id := range c.getIdentifiersFromAssoc(newPod) {
 		// compare initial scheduled timestamp for existing pod and new pod with same identifier
-		// and only replace old pod if scheduled time of new pod is newer or equal.
+		// and only replace old pod if scheduled time of new pod is newer.
```
Comment (evan-bradley) [inline processor/k8sattributesprocessor/internal/kube/client.go:522]: The old comment should still stand here: the pod object will get replaced if the new pod has a starting time equal to or greater than that of the old pod.

**Review by evan-bradley — COMMENTED** [review]
I tested this on a k3d cluster and it fixes the issue, nice work. Could you add a unit test to cover this case and make sure this change has a changelog entry?

File: processor/k8sattributesprocessor/internal/kube/client.go (hunk 1 above)
Comment (willthames) [reply processor/k8sattributesprocessor/internal/kube/client.go:522]: thanks, good point!

**Comment by willthames** [comment]
> Could you add a unit test to cover this case?

Genuinely struggling with this one, any advice gratefully received. It seems I don’t understand the cause as well as I know the fix. My first attempt added start date to podC in TestPodAdd but that didn’t fail. I’ve tried adding extra associations but that just caused a bunch of seemingly unrelated failures

**Comment by willthames** [comment]
That's great, thanks @evan-bradley - I hadn't realised that there were three events. I'll write a test that reflects that.

**Comment by willthames** [comment]
I wrote the test and ensured it failed and then cherry-picked it onto the branch and it then worked - I've left the commits as separate if you want to validate the test fails without the change.

**Review by evan-bradley — COMMENTED** [review]
Overall looks good, just some minor comments.

**Review by evan-bradley — APPROVED** [review]
Looks good! Thanks for fixing this.

**Comment by evan-bradley** [comment]
@dmitryax Could you take a look at this?

File: processor/k8sattributesprocessor/internal/kube/client.go [hunk 2]
```diff
@@ -523,7 +523,7 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 		// This should fix the case where scheduler has assigned the same attribtues (like IP address)
 		// to a new pod but update event for the old pod came in later.
 		if p, ok := c.Pods[id]; ok {
-			if p.StartTime != nil && !p.StartTime.Before(pod.Status.StartTime) {
+			if p.StartTime != nil && !p.StartTime.Before(pod.Status.StartTime) && !p.StartTime.Equal(pod.Status.StartTime) {
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/internal/kube/client.go:526]: wouldn't this be the same?
```suggestion
			if pod.Status.StartTime.Before(p.StartTime) {
```

**Comment by willthames** [comment]
@dmitryax you're quite right - I tested it with your much more readable suggestion, and all tests pass, so I've committed your suggestion, thanks!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**